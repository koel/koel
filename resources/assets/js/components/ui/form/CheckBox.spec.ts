import { describe, expect, it } from 'vitest'
import { screen } from '@testing-library/vue'
import { createHarness } from '@/__tests__/TestHarness'
import Component from './CheckBox.vue'

describe('checkBox.vue', () => {
  const h = createHarness()

  it('renders unchecked state', () => expect(h.render(Component).html()).toMatchSnapshot())

  it('renders checked state', () => expect(h.render(Component, {
    props: {
      modelValue: true,
    },
  }).html()).toMatchSnapshot())

  it('emits the input event', async () => {
    const { emitted } = h.render(Component)

    await h.trigger(screen.getByRole('checkbox'), 'click')

    expect(emitted()['update:modelValue']).toBeTruthy()
  })
})
