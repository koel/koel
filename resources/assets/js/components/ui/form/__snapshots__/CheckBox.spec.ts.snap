// Vitest Snapshot v1

exports[`checkBox.vue > renders checked state 1`] = `<input checked="" class="relative align-bottom inline-block w-[32px] h-[20px] bg-gray-400 rounded-full shadow-inner cursor-pointer transition-all duration-200 ease-in-out mr-2 after:h-[16px] after:aspect-square after:absolute after:bg-white after:top-[2px] after:left-[2px] after:rounded-full after:transition-left after:duration-200 after:ease-in-out checked:bg-k-highlight checked:after:left-[14px]" type="checkbox">`;

exports[`checkBox.vue > renders unchecked state 1`] = `<input class="relative align-bottom inline-block w-[32px] h-[20px] bg-gray-400 rounded-full shadow-inner cursor-pointer transition-all duration-200 ease-in-out mr-2 after:h-[16px] after:aspect-square after:absolute after:bg-white after:top-[2px] after:left-[2px] after:rounded-full after:transition-left after:duration-200 after:ease-in-out checked:bg-k-highlight checked:after:left-[14px]" type="checkbox">`;
